service/create/launch_configuration.go

```diff
@@ -91,21 +93,23 @@ func (s *Service) createLaunchConfiguration(input launchConfigurationInput) (boo
 		return false, microerror.MaskAny(err)
 	}
 
-	launchConfigName, err := launchConfigurationName(input.cluster, input.prefix)
+	launchConfigName, err := launchConfigurationName(input.cluster, input.prefix, securityGroupID)
 	if err != nil {
 		return false, microerror.MaskAny(err)
 	}
 
 	launchConfig := &awsresources.LaunchConfiguration{
 		Client: input.clients.AutoScaling,
 		Name:   launchConfigName,
-		IamInstanceProfileName: input.instanceProfileName,
-		ImageID:                imageID,
-		InstanceType:           instanceType,
-		KeyName:                input.keypairName,
-		SecurityGroupID:        securityGroupID,
-		SmallCloudConfig:       smallCloudconfig,
+		IamInstanceProfileName:   input.instanceProfileName,
+		ImageID:                  imageID,
+		InstanceType:             instanceType,
+		KeyName:                  input.keypairName,
+		SecurityGroupID:          securityGroupID,
+		SmallCloudConfig:         smallCloudconfig,
+		AssociatePublicIpAddress: publicIP,
```

Comment:
Yes these inputs have a lot of fields so alphabetical order makes sense. If we're doing that we should do it everywhere so please raise an issue to tidy this up.